### PR TITLE
fire service protocol events for frames

### PR DESCRIPTION
**Pull request by devoncarew**
- redux of https://github.com/flutter/flutter/pull/10966 - fire service protocol events for frames. This fires `Flutter.Frame` events to the service protocol, in the format `{'number': ..., 'startTime': ..., 'elapsed': ...}`
- add a `service_extensions_test.dart` test, and refactor the devicelab tests a bit to pull the service protocol scraping code into common utils

### CODE REVIEW

File: dev/devicelab/bin/tasks/commands_test.dart [hunk 1]
```diff
@@ -38,10 +35,9 @@ void main() {
         .listen((String line) {
           print('run:stdout: $line');
           stdout.add(line);
-          if (line.contains(observatoryRegExp)) {
-            final Match match = observatoryRegExp.firstMatch(line);
-            vmServicePort = int.parse(match.group(2));
-            print('service protocol connection available at ${match.group(1)}');
+          if (lineContainsServicePort(line)) {
+            vmServicePort = parseServicePort(line);
```
Comment (Hixie) [inline dev/devicelab/bin/tasks/commands_test.dart]: This applies the regexp twice, which isn't ideal.

**Comment by Hixie** [comment]
LGTM

**Comment by Hixie** [comment]
@devoncarew What's the status of this PR?

**Comment by devoncarew** [comment]
Sorry - will land this, I just need to find a bit of time when I can triage bots if they fail.

**Comment by devoncarew** [comment]
This is causing failures in the bots that run in profile mode (flutter run --profile). The errors we're seeing are:

```
E/DartVM  (13226): ../../dart/runtime/vm/object.cc: 13854: error: unreachable code
E/DartVM  (13226): Dumping native stack trace for thread 33cc
E/DartVM  (13226):   [0xd2385159] Unknown symbol
E/DartVM  (13226):   [0xd2385159] Unknown symbol
E/DartVM  (13226): -- End of DumpStackTrace
F/libc    (13226): Fatal signal 6 (SIGABRT), code -6 in tid 13260 (ui_thread)
```

I think this is caused by calling the dart:developer `postEvent()` method when the app is compiled in profile mode. I think this should be supported, but may never have been called before. /cc @a-siva 

I'll revert this PR and open an issue for the crash in postEvent() in profile mode.

**Comment by a-siva** [comment]
Which tool is going to use the 'Flutter.Frame' event ?

**Comment by devoncarew** [comment]
I have a prototype UI (https://github.com/flutter/flutter/pull/10966#issuecomment-316250023) showing frame event info in IntelliJ. It's likely we'd end up using this info in other tools though, once it's more generally available.

**Pull request merged**